federation/pkg/federation-controller/ingress/ingress_controller.go

```diff
@@ -287,6 +297,9 @@ func (ic *IngressController) reconcileIngress(ingress types.NamespacedName) {
 			// We can't supply server-created fields when creating a new object.
 			desiredIngress.ObjectMeta.ResourceVersion = ""
 			desiredIngress.ObjectMeta.UID = ""
+			ic.eventRecorder.Eventf(baseIngress, api.EventTypeNormal, "CreateInCluster",
+				"Creating ingress in cluster %s", cluster.Name)
```

Comment:
The event is attached to the ingress object and all of the stuff is there. We don't include event owner in the event text.
Example:
https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/controller_utils.go#L479